src/query/storage/m3/cluster_test.go

```diff
@@ -152,15 +154,87 @@ func TestNewClustersFromConfig(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func TestNewDynamicClusters(t *testing.T) {
+	ctrl := xtest.NewController(t)
+	defer ctrl.Finish()
+
+	newClient1, session1, client1 := newTestClientFromConfigWithNamespaceInitializer(ctrl, true)
+	newClient2, session2, client2 := newTestClientFromConfigWithNamespaceInitializer(ctrl, true)
+	cfg := ClustersStaticConfiguration{
+		ClusterStaticConfiguration{
+			NewClientFromConfig: newClient1,
+		},
+		ClusterStaticConfiguration{
+			NewClientFromConfig: newClient2,
+		},
+	}
+
+	clusters, err := cfg.newDynamicClusters(
+		newNoopCluster,
```

Comment:
nit: would it be better to generate a `Clusters` mock? Your call

Reply:
Eh, probably, but I think this test will change a bit once I'm wiring up, so will leave as-is for now